### DISCUSSION TITLE
GitHub Copilot could not connect to the server

### DISCUSSION THREAD

**Discussion by azizijunaid** [community/community#11016] [Copilot Conversations]
**When i try to connect it is showing an error**

`GitHub Copilot could not connect to the server. Extension activation failed: "Timed out waiting for authentication provider to register`

**Comment by davecheney** [comment]
Hello, I'm sorry you are having difficulties. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by azizijunaid** [reply]
No there is no firewall and vpn but it is working fine in windows but having this issue in macbook

**Reply by davecheney** [reply]
Thanks for confirming. This sounds like a local problem on your machine. I'm sorry, I don't think I can provide more assistance other than suggesting

<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by azizijunaid** [reply]
I signed in with a Github account but it shows in brackets that I am logged in with Microsoft, aaahhhh i am frustrated with this problem.

**Reply by fernandopoli** [reply]
I am having the same issue.

**Reply by faruk56-arch** [reply]
i am having same problem dont know why

**Reply by davecheney** [reply]
@fernandopoli @faruk56-arch sorry youre having problems.

<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

if that does not work, can you please describe exactly what you are seeing, screenshots or please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by Taraldinn** [reply]
![image](https://user-images.githubusercontent.com/59772487/153477053-7d4a41e5-46f4-42fd-b131-26e3b838ca7f.png)
# I got same problem what to do ?
@davecheney

**Reply by davecheney** [reply]
@programmer-Quazi is it possible that your employer, internet service provider, or government is intercepting your network traffic?

**Comment by azizijunaid** [comment]
I have resolved this by removing vscode data.

**Reply by faruk56-arch** [reply]
I got the solution by adding the GitHub in va code
Best of luck

**Reply by fernandopoli** [reply]
@azizijunaid how did you do that?

**Reply by azizijunaid** [reply]
I have resolved by removing vscode data from laptop

**Comment by Jeong-Min-Cho** [comment]
I had this problem with my Ubuntu system.
But, I resolved the issue by deleting the vscode configs.

![image](https://user-images.githubusercontent.com/9837449/169903190-6aad5262-2d81-4d94-b585-909cc1fe0aec.png)
Source: https://code.visualstudio.com/docs/setup/uninstall

For me, deleting ```$HOME/.config/Code``` is enough to prompt the Github sign-in again by using ```rm -rf``` command.

**WARNING: This may remove all your extension and VSCode settings !!**

Hopefully, this will help someone in the future.

**Comment by GoulartNogueira** [comment]
For me, the root cause was:

- I had not approved the current workspace as a secure environment.
- Somehow the github cannot login in that scenario
- I recieved the same error message

I tried everything (including resetting all the config)

**How I fixed:**

- Simple change to another workspace.
- Confirm it's a secure workspace
- Automatic login to github
- Copilot working perfectly
